File: JavaScript/shouyv2.js
/*
  这个应该很像兽音译者了，尽管加密不同
*/

/*function to2core(data){
    let list = []
    for(let i=0;i<data.length;i++){
        list.push(data[i].charCodeAt().toString(2))
    }   
    return list.join("");
}
console.log(to2core("₠"))
/*function tostring(data) {  
    // 将二进制字符串转换为 ArrayBuffer，然后创建一个 Uint8Array 视图来访问它  
    const length = data.length;  
    if (length % 8 !== 0) {  
        return false  
    }  
    const byteArray = new Uint8Array(new ArrayBuffer(length / 8));  
    // 遍历二进制字符串，每 8 位一组转换为 Uint8Array 中的一个字节  
    for (let i = 0; i < length; i += 8) {  
        // 使用 parseInt 将 8 位二进制字符串转换为十进制数（字节）  
        const byte = parseInt(data.substr(i, 8), 2);  
        byteArray[i / 8] = byte;  
    }  
    // 使用 Buffer.from 创建一个新的 Buffer 实例，该实例包含与 Uint8Array 相同的数据  
    const buffer = Buffer.from(byteArray.buffer);  
    // 最后，将 Buffer 解码为字符串（假设它是 UTF-8 编码的）  
    return buffer.toString('utf8');  
}  */
/*function tostring(data){
  
}*/
//console.log(tostring("111000101000001010100000"))
//console.log(" ".charCodeAt().toString(16))
//console.log(0x10)
//console.log(String.fromCharCode(0x4B))

//[["嗷"],["呜"],["啊"],["~"]]
function infurr(input,sign) {
  function getMv(left,right,sign) {
  function random(data){
    //console.log(data)
    return data[Math.floor(data.length*Math.random())]
  }
  //console.log(sign)
  //console.log()
  return [random(sign[Math.floor(parseInt(`0x${left}`)/4)])+random(sign[parseInt(`0x${left}`)%4+parseInt(`0x${right}`)%4]),random(sign[Math.floor(parseInt(`0x${right}`)/4+Math.floor(parseInt(`0x${left}`)/4))])+random(sign[parseInt(`0x${right}`)%4])]
}
  function add0(str){
    return ((str.length===2)&&`00${str}`)||((str.length===3)&&`0${str}`)||str
  }
  sign.push(...sign)
  let list_one=[],list_two=[],data=[]
  for (let i=0;i<input.length;i++){
    data.push(add0(input[i].charCodeAt(0).toString(16)))
  }
  //console.log(data)
  data=data.join("")
  //console.log(data)
  for(let i=0,len=data.length;i<len/2;i++){
    //console.log(data[i],data[len-1-i])
    list_one.push(getMv(data[i],data[len-1-i],sign)[0])
    //console.log(getMv(data[i],data[len-1-i],sign)[0])
    list_two.unshift(getMv(data[i],data[len-1-i],sign)[1])
  }
  list_one.push(...list_two)
  //console.log(list_one)
  return list_one.join("")
}

let signn=[["嗷","喵"],["呜"],["啊"],["~"]]
//console.log(infurr("7",signn))

function outfurr(input,sign){
  function getMc(left,right,sign){
  let d=[parseInt(sign[left[0]]),parseInt(sign[left[1]]),parseInt(sign[right[0]]),parseInt(sign[right[1]])]
  //console.log(d,(d[0]*4+d[1]-d[3]+((d[1]-d[3]<0)&&4||0)).toString(16),((d[2]-d[0]+((d[2]-d[0]<0&&4)||0))*4+d[3]).toString(16))
  return [(d[0]*4+d[1]-d[3]+((d[1]-d[3]<0)&&4||0)).toString(16),((d[2]-d[0]+((d[2]-d[0]<0&&4)||0))*4+d[3]).toString(16)]
}
  let resign ={},list_one=[],list_two=[],list=[]
  for (let i=0;i<sign.length;i++){
    for (let ele of sign[i]) {
      resign[ele]=String(i)
    }
  }
  //console.log(resign)
  for (let i=0,len=input.length;i<len/2;i+=2){
    //console.log(input.slice(i,i+2),input.slice(-i-2+len,len-i))
    list_one.push(getMc(input.slice(i,i+2),input.slice(-i-2+len,len-i),resign)[0])
    list_two.unshift(getMc(input.slice(i,i+2),input.slice(-i-2+len,len-i),resign)[1])
  }
  //console.log(list_one)
  list_one.push(...list_two)
  //console.log(list_one)
  for (let i =0;i<list_one.length;i+=4) {
    //console.log(parseInt(list_one.slice(i,i+4).join(""),16))
    list.push(String.fromCharCode(parseInt(list_one.slice(i,i+4).join(""),16)))
  }
  return list.join("")
}
/*function getMc(left,right,sign){
  let d=[parseInt(sign[left[0]]),parseInt(sign[left[1]]),parseInt(sign[right[0]]),parseInt(sign[right[1]])]
  //console.log(d,(d[0]*4+d[1]-d[3]+((d[1]-d[3]<0)&&4||0)).toString(16),((d[2]-d[0]+((d[2]-d[0]<0&&4)||0))*4+d[3]).toString(16))
  return [(d[0]*4+d[1]-d[3]+((d[1]-d[3]<0)&&4||0)).toString(16),((d[2]-d[0]+((d[2]-d[0]<0&&4)||0))*4+d[3]).toString(16)]
}*/
//console.log(outfurr("嗷啊呜啊啊啊嗷喵",signn))
console.log(infurr("①",signn))

File: python/shouyv2.py
"""
  写成shit恶心人
"""

import random
def infurr(_input,_sign=None,d={},mode=None):
  return (mode==None and infurr(_input,_sign+_sign,{"output_":[],"i_":0},"for_common"))or(mode=="for_common" and ((d["i_"] == len(_input) and infurr("".join(d["output_"]),_sign,{"output_":[],"i_":0},"for_add_mi")) or infurr(_input,_sign,{"output_":d["output_"]+[infurr(hex(ord(_input[d["i_"]]))[2:],None,None,"add0")],"i_":d["i_"]+1}, mode)))or(mode=="for_add_mi" and ((d["i_"] == int(len(_input) / 2) and "".join(d["output_"])) or infurr(_input,_sign,{"output_":d["output_"][:d["i_"]]+infurr({"left":_input[d["i_"]], "right":_input[len(_input) - 1 - d["i_"]]}, _sign,None,"getMv")+d["output_"][d["i_"]:], "i_":d["i_"] + 1},mode)))or(mode=="add0" and (((len(_input) == 2) and f"00{_input}") or ((len(_input) == 3) and f"0{_input}") or _input))or(mode=="getMv" and([infurr(_sign[int(int(_input["left"],16) / 4)],None,None,"random") + infurr(_sign[int(_input["left"],16) % 4 + int(_input["right"],16) % 4],None,None,"random"),infurr(_sign[int(int(_input["right"],16) / 4 + int(int(_input["left"],16) / 4))],None,None,"random") + infurr(_sign[int(_input["right"],16) % 4],None,None,"random")]))or(mode=="random" and _input[random.randint(0,len(_input)-1)])
print(infurr("114514",[["喵", "嗷"], ["呜"], ["~"], ["咪"]]))
"""def outfurr(_input,_sign=None,d={},mode=None):
  return (mode==None and outfurr(_input,outfurr(_sign,None,{"output_":{},"i_1":0,"i_2":0},"getSign"),{"output_":[],"i_":0},"for_del_mi"))
  (mode=="getSign" and ((d["i_1"] == 4 and d["output_"]) or (len(_input[d["i_1"]]) > d["i_2"] and outfurr(_input,None,{"output_":{**d["output_"],_input[d["i_1"]][d["i_2"]]:d["i_1"]},"i_1":d["i_1"],"i_2":d["i_2"]+1},"getSign")) or outfurr(_input,None,{"output_":d["output_"],"i_1":d["i_1"]+1,"i_2":0},"getSign")))
  (mode=="for_del_mi" and ((d["i_"] == len(_input) / 2 and outfurr(d["output_"],None,{"output_":[],"i_":0},"for_uncommon")) or outfurr(_input,_sign,{"output_":d["output_"][:int(d["i_"] / 2)]+outfurr({"left":_input[d["i_"]:d["i_"] + 2], "right":_input[-d["i_"] - 2 + len(_input):len(_input) - d["i_"]]}, _sign,None,"getMc")+d["output_"][int(d["i_"]/2):],"i_":d["i_"] + 2}, "for_del_mi")))
  #outfurr({"left":.,"right":},_sign,None,"getMc")
  (mode=="getMc" and [hex(int(_sign[_input["left"][0]]) * 4 + int(_sign[_input["left"][1]]) - int(_sign[_input["right"][1]]) + ((int(_sign[_input["left"][1]]) - int(_sign[_input["right"][1]]) < 0) and 4 or 0))[2:],hex((int(_sign[_input["right"][0]]) - int(_sign[_input["left"][0]]) + ((int(_sign[_input["right"][0]]) - int(_sign[_input["left"][0]]) < 0 and 4) or 0)) * 4 + int(_sign[_input["right"][1]]))[2:]])
  #outfurr(_input,None,{"output_":[],"i_":0},"for_uncommon")
  (mode=="for_uncommon" and ((d["i_"] == len(_input) and d["output_"]) or outfurr(_input,None,{"output_":d["output_"]+[chr(int("".join(_input[d["i_"]: d["i_"] + 4]), 16))],"i_": d["i_"] + 4}, "for_uncommon")))
"""
def outfurr(_input,_sign=None,d={},mode=None):
  return (mode==None and outfurr(_input,outfurr(_sign,None,{"output_":{},"i_1":0,"i_2":0},"getSign"),{"output_":[],"i_":0},"for_del_mi"))or(mode=="for_del_mi" and ((d["i_"] == len(_input) / 2 and outfurr(d["output_"],None,{"output_":[],"i_":0},"for_uncommon")) or outfurr(_input,_sign,{"output_":d["output_"][:int(d["i_"] / 2)]+outfurr({"left":_input[d["i_"]:d["i_"] + 2], "right":_input[-d["i_"] - 2 + len(_input):len(_input) - d["i_"]]}, _sign,None,"getMc")+d["output_"][int(d["i_"]/2):],"i_":d["i_"] + 2}, "for_del_mi")))or(mode=="for_uncommon" and ((d["i_"] == len(_input) and "".join(d["output_"])) or outfurr(_input,None,{"output_":d["output_"]+[chr(int("".join(_input[d["i_"]: d["i_"] + 4]), 16))],"i_": d["i_"] + 4}, "for_uncommon")))or(mode=="getSign" and ((d["i_1"] == 4 and d["output_"]) or (len(_input[d["i_1"]]) > d["i_2"] and outfurr(_input,None,{"output_":{**d["output_"],_input[d["i_1"]][d["i_2"]]:d["i_1"]},"i_1":d["i_1"],"i_2":d["i_2"]+1},"getSign")) or outfurr(_input,None,{"output_":d["output_"],"i_1":d["i_1"]+1,"i_2":0},"getSign")))or(mode=="getMc" and [hex(int(_sign[_input["left"][0]]) * 4 + int(_sign[_input["left"][1]]) - int(_sign[_input["right"][1]]) + ((int(_sign[_input["left"][1]]) - int(_sign[_input["right"][1]]) < 0) and 4 or 0))[2:],hex((int(_sign[_input["right"][0]]) - int(_sign[_input["left"][0]]) + ((int(_sign[_input["right"][0]]) - int(_sign[_input["left"][0]]) < 0 and 4) or 0)) * 4 + int(_sign[_input["right"][1]]))[2:]])
print(outfurr("喵嗷喵咪喵咪嗷呜嗷呜喵咪嗷咪喵呜嗷呜喵咪喵咪呜嗷呜喵喵嗷喵咪呜呜嗷嗷喵嗷喵咪喵呜嗷喵嗷嗷嗷咪呜嗷",[["喵", "嗷"], ["呜"], ["~"], ["咪"]]))
